src/bindings/python/src/pyopenvino/frontend/node_context.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "node_context.hpp"
+
+#include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+#include "openvino/frontend/node_context.hpp"
+
+namespace py = pybind11;
+
+using namespace ov::frontend;
+
+template <typename>
+struct is_std_vector : std::false_type {};
+
+template <typename T, typename A>
+struct is_std_vector<std::vector<T, A>> : std::true_type {};
+
+#define CAST_VEC_TO_PY(any, py_type, c_type)                                       \
+    {                                                                              \
+        static_assert(is_std_vector<c_type>(), "The type should be std::vector."); \
+        if ((any).is<c_type>()) {                                                  \
+            auto casted = (any).as<c_type>();                                      \
+            if (!(py_type).is_none()) {                                            \
+                py::list py_list;                                                  \
+                for (const auto& el : casted) {                                    \
+                    py_list.append(py_type(el));                                   \
+                }                                                                  \
+                return py_list;                                                    \
+            }                                                                      \
+            return py::cast(casted);                                               \
+        }                                                                          \
+    }
+
+#define CAST_TO_PY(any, py_type, c_type)      \
```

Comment:
In both of those macros the `py_type` is in fact a `py::object`. In the return statement it's used as if it was a type: `return py_type(casted)`. What does it do in practice? Just wraps `any` casted to a C++ type in a `py::object`?
I'm asking because I'm thinking about a way of changing those macros to some templated code instead.

Reply:
What does it do in practice? 
Calls ctor of python object, it can be built-in classes as int,float and specified by user, e.g. numpy.float32

Reply:
discussed offline.
decided to leave it as is